File: Golang/reverse-integer.go
import "math"
func reverse(x int) int {

    reversed_x := 0
    for x != 0 {
        reversed_x = reversed_x * 10 + x % 10
        x /= 10
    }

    if math.MinInt32 > reversed_x || reversed_x > math.MaxInt32 {
        return 0
    } else {
        return reversed_x
    }
}

File: Python3/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        multiplier = -1 if x < 0 else 1

        x *= multiplier
        reversed_x = 0
        while x != 0:
            reversed_x = reversed_x * 10 + x % 10
            x //= 10

            if reversed_x >= 2**31:
                return 0

        return reversed_x * multiplier
